codalab/lib/beam/blobstorageuploader.py

```diff
@@ -51,5 +51,8 @@ def _write_to_blob(self, data):
     self.block_number = self.block_number + 1
 
   def finish(self):
-    self._write_to_blob(self.buffer)
+    while len(self.buffer) > 0:
```

Comment:
why is `finish()` being called if there are still more bytes in the buffer than `MAX_WRITE_SIZE`?

Reply:
Ok, that's true -- there will always be less bytes in the buffer than `MIN_WRITE_SIZE` at this point. Will change it to just do:

`self._write_to_blob(self.buffer.read())`

instead.